src/components/Layouts/Shared/getNavPages.tsx

```diff
@@ -0,0 +1,223 @@
+import { LinkProps } from 'next/link';
+import { ReactElement, useMemo } from 'react';
+import CompassIcon from '@mui/icons-material/Explore';
+import { useTranslation } from 'react-i18next';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { reportNavItems } from './reportNavItems';
+import { settingsNavItems } from './settingsNavItems';
+import { toolsNavItems } from './toolsNavItems';
+
+interface SubItems {
+  id?: string;
+  title?: string;
+  grantedAccess?: string[];
+}
+interface Item {
+  id?: string;
+  href?: LinkProps['href'];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+  subtitle?: string;
+  desc?: string;
+  // search dialog specific
+  showInSearchDialog?: boolean;
+  searchIcon?: ReactElement;
+  searchName?: string;
+  // settings specific
+  grantedAccess?: string[];
+  subItems?: SubItems[];
+  oauth?: boolean;
+}
+
+export interface NavPage {
+  id?: string;
+  title: string;
+  href?: LinkProps['href'];
+  pathname?: string;
+  searchIcon?: ReactElement;
+  items?: Item[];
+  whatsNewLink?: boolean;
+  showInNav?: boolean;
+  showInSearchDialog?: boolean;
+  showInPanel?: boolean;
+}
+
+export function getNavPages(
+  coachingAccountCount: number | undefined,
+  isSearch = false,
+) {
+  const accountListId = useAccountListId();
+  const { t } = useTranslation();
+
+  const allNavPages = useMemo<NavPage[]>(() => {
+    const navPages: NavPage[] = [
+      {
+        title: t('Dashboard'),
+        href: `/accountLists/${accountListId}`,
+        pathname: '/accountLists/[accountListId]',
+        showInNav: true,
+      },
+      {
+        title: t('Contacts'),
+        searchIcon: <CompassIcon />,
+        href: `/accountLists/${accountListId}/contacts`,
+        pathname: '/accountLists/[accountListId]/contacts/[[...contactId]]',
+        showInNav: true,
+        showInSearchDialog: true,
+      },
+      {
+        title: t('Tasks'),
+        searchIcon: <CompassIcon />,
+        href: `/accountLists/${accountListId}/tasks`,
+        pathname: '/accountLists/[accountListId]/tasks/[[...contactId]]',
+        showInNav: true,
+        showInSearchDialog: true,
+      },
+      {
+        id: 'reports-page',
+        title: t('Reports'),
+        pathname: '/accountLists/[accountListId]/reports',
+        items: reportNavItems.map((item) => ({
```

Comment:
When building up the items, would it make sense to spread `...item` instead of manually copying id, title, and subtitle, for example?